python/lsst/summit/utils/guiders/seeing.py

```diff
@@ -0,0 +1,518 @@
+                corrs.append([np.nan, np.nan])
+                covs.append([np.nan, np.nan])
+                diffVars.append([np.nan, np.nan])
+                continue
+
+            c1, c2 = _centers(df1, df2)
+            cent1 = _correctedXy(df1)
+            cent2 = _correctedXy(df2)
+            rot1, rot2 = rotateCentroids(cent1, cent2, c1, c2)
+
+            pairCorrs: list[float] = []
+            pairCovs: list[float] = []
+            pairDvars: list[float] = []
+            for comp in (0, 1):  # 0 = ⟂ (x), 1 = ∥ (y)
+                corr, cov, dvar = _corrCovDiff(rot1[i1, comp], rot2[i2, comp])
+                pairCorrs.append(corr)
+                pairCovs.append(cov)
+                pairDvars.append(dvar)
+
+            corrs.append(pairCorrs)
+            covs.append(pairCovs)
+            diffVars.append(pairDvars)
+
+        return (
+            np.asarray(corrs, dtype=np.float64),
+            np.asarray(covs, dtype=np.float64),
+            np.asarray(diffVars, dtype=np.float64),
+        )
+
+    def measureTomoSeeing(self) -> tuple[float, float, float, float]:
+        """
+        Estimate tomographic seeing by splitting the total FWHM into
+        low/mid/high altitude contributions.
+
+        This method:
+        1) computes pairwise correlations for three detector-pair presets
+           (adjacent, cross, diagonal),
+        2) estimates a total Fried parameter from the median centroid
+           variance across detectors, and
+        3) converts to total FWHM using a Vonkármán model and scales to
+           the exposure wavelength,
+        4) splits the total into low/mid/high layers using empirical
+           weights based on correlations.
+
+        Returns
+        -------
+        fwhmTot : `float`
+            Total atmospheric FWHM (arcsec) at the exposure wavelength.
+        fwhmLow : `float`
+            Low-altitude contribution to the FWHM (arcsec).
+        fwhmMid : `float`
+            Mid-altitude contribution to the FWHM (arcsec).
+        fwhmHigh : `float`
+            High-altitude contribution to the FWHM (arcsec).
+        """
+        corrsAdj, _, _ = self.measureCorrPairs("adjacent")
+        corrsCross, _, _ = self.measureCorrPairs("cross")
+        corrsDiag, _, _ = self.measureCorrPairs("diagonal")
+        var = self.measureVar()
+
+        r0 = r0FromVar(float(np.nanmedian(var)))
+        # Wavelength scaling: r0 ∝ λ^(6/5) ⇒ FWHM ∝ λ^(-1/5)
+        fwhmTot = float(fwhmVonkarman(r0) * (self.wavelength / 500.0) ** (-1.0 / 5.0))
+
+        weightsLow = (
+            0.332 * float(np.nanmedian(corrsDiag[:, 1]))
+            + 0.21 * float(np.nanmedian(corrsCross[:, 0]))
+            + 0.458 * float(np.nanmedian(corrsDiag[:, 0]))
+        )
+        weightsHigh = (
+            1.0 - 0.872 * float(np.nanmedian(corrsAdj[:, 0])) - 0.128 * float(np.nanmedian(corrsAdj[:, 1]))
+        )
+        weightsMid = 1.0 - weightsLow - weightsHigh
+
+        return (
```

Comment:
Consider a dataclass here.

`TomogrophicSeeing` with variables: total, low, middle, high. Then, the usage would be:

```
seeing = analysis.measureTomoSeeing()
seeing.total # print the total seeing
seeing.low # ...
```

Reply:
Tip: don't forget to change the type on the top of the method
`def measureTomoSeeing(self) -> TomogrophicSeeing:`

Also, if you want to learn more about dataclasses, take a look at this nice video:
[Arjan Dataclass](https://youtu.be/CvQ7e6yUtnw?si=pfKS4pannmM-5xYf)